src/Amplitude.php

```diff
@@ -0,0 +1,498 @@
+<?php
+namespace Zumba\Amplitude;
+
+class Amplitude
+{
+    const AMPLITUDE_API_URL = 'https://api.amplitude.com/httpapi';
+
+    const EXCEPTION_MSG_NO_API_KEY = 'API Key is required to log an event';
+    const EXCEPTION_MSG_NO_EVENT_TYPE = 'Event Type is required to log or queue an event';
+    const EXCEPTION_MSG_NO_USER_OR_DEVICE = 'Either user_id or device_id required to log an event';
+
+    /**
+     * The API key to use for all events generated by this instance
+     *
+     * @var string
+     */
+    protected $apiKey;
+
+    /**
+     * The event that will be used for the next event being tracked
+     *
+     * @var \Zumba\Amplitude\Event
+     */
+    protected $event;
+
+    /**
+     * The user ID to use for events generated by this instance
+     *
+     * @var string
+     */
+    protected $userId;
+
+    /**
+     * The user data to set on the next event logged to Amplitude
+     *
+     * @var array
+     */
+    protected $userProperties = [];
+
+    /**
+     * The device ID to use for events generated by this instance
+     *
+     * @var string
+     */
+    protected $deviceId;
+
+    /**
+     * Queue of events, used to allow generating events that might happen prior to amplitude being fully initialized
+     *
+     * @var \Zumba\Amplitude\Event[]
+     */
+    protected $queue = [];
+
+    /**
+     * Flag for if user is opted out of tracking
+     *
+     * @var boolean
+     */
+    protected $optOut = false;
+
+    /**
+     * Array of Amplitude instances
+     *
+     * @var \Zumba\Amplitude\Amplitude[]
+     */
+    private static $instances = [];
+
+    /**
+     * Singleton to get named instance
+     *
+     * Using this is optional, it depends on the use-case if it is better to use a singleton instance or just create
+     * a new object directly.
+     *
+     * Useful if want to possibly send multiple events for the same user in a single page load, or even keep track
+     * of multiple named instances, each could track to it's own api key and/or user.
+     *
+     * Example:
+     * <code>
+     * // In user initialization section of app...
+     * Zumba\Amplitude\Amplitude::getInstance()->init('APIKEY','johnny@example.com')
+     *     ->addUserProperties(['name' => 'Johnny 5'])
+     *     ->logQueuedEvents();
+     *
+     * // Elsewhere in your app, this could happen before OR after the above initialization...
+     * // See queueEvent() for more info
+     * Zumba\Amplitude\Amplitude::getInstance->queueEvent(
```

Comment:
@jonyo should it be `Zumba\Amplitude\Amplitude::getInstance()->queueEvent(` (missing `()`)